test/unit/DCAKeep3rJob/dca-keep3r-job.spec.ts

```diff
@@ -97,4 +98,41 @@ contract('DCAKeep3rJob', () => {
       });
     });
   });
+  describe('setSwapper', () => {
+    when('zero address is sent', () => {
+      then('reverts with message', async () => {
+        await behaviours.txShouldRevertWithMessage({
+          contract: DCAKeep3rJob.connect(superAdmin),
+          func: 'setSwapper',
+          args: [constants.ZERO_ADDRESS],
+          message: 'ZeroAddress',
+        });
+      });
+    });
+    when('a valid address is sent', () => {
+      const NEW_SWAPPER = constants.NOT_ZERO_ADDRESS;
+      let tx: TransactionResponse;
+      given(async () => {
+        tx = await DCAKeep3rJob.connect(superAdmin).setSwapper(NEW_SWAPPER);
+      });
+      then('it is set correctly', async () => {
+        const { swapper } = await DCAKeep3rJob.swapperAndNonce();
+        expect(swapper).to.equal(NEW_SWAPPER);
+      });
+      then('nonce is not modified', async () => {
+        const { nonce } = await DCAKeep3rJob.swapperAndNonce();
+        expect(nonce).to.equal(constants.ZERO);
+      });
```

Comment:
Should we test this with `nonce != 0`? Although... if nonce is changed when swapper gets changed... It's an issue with the EVM, not with our code lmao.

Reply:
Not sure I understand the suggestion. It was zero before the swapper was set, want to check it's still zero after it's set